### PR TITLE
enable email relay for users who have not opted out

### PR DESCRIPTION
**Pull request by nburt**

### CODE REVIEW

**Comment by nburt** [comment]
I've already deployed the function to prod + run reloaddatasheet on each country so that the encrypted email field is stripped

**Comment by linderr** [comment]
May we talk about this before merging??? Especially w/the data leads @4kbt @cchang-adi

**Comment by 4kbt** [comment]
Let's wait until the morning (see discussion in #dev) to merge. We want to be real sure that we don't antagonize recipients :).

File: functions/index.js [hunk 1]
```diff
@@ -334,13 +338,25 @@ async function snapshotData(country) {
...
+    const contactFormOptOutIndex = orig_col_labels.findIndex(e => e === '_contact_form_opt_out');
+
     const trimmed_values = real_values.map((row, row_num) => {
       const result = [];
 
+      // Opt out of mail relay if
+      // country's sheet does not have opt out option
+      // OR opt out column has content in it
+      const hasOptedOutOfContact = contactFormOptOutIndex === -1
+        || (Boolean(row[contactFormOptOutIndex]) && row[contactFormOptOutIndex].trim().length > 0);
```
Comment (awong-dev) [inline functions/index.js:350]: How can (Boolean(row[contactFormOptOutIndex]) be true
and row[contactFormOptOutIndex].trim().length be 0?

Comment (nburt) [reply functions/index.js:350]: The index === -1 check might have fixed this... I can revisit but originally was checking to make sure that we weren't calling `trim().length` on undefined but we might not need that.

Ultimately I think this field should be explicit as an enum with 'Yes' / 'No'. We can migrate to that by:

1. Replacing the current opt out text with "Yes" but leaving the no's as blank
2. Update functions/index.js to check for presence + value === 'Yes'
3. Update form to write No answers too

**Pull request merged**